common/workunit/workunit.cpp

```diff
@@ -6359,6 +6429,83 @@ unsigned CLocalWorkUnit::queryFileUsage(const char *fileName) const
     return p->getPropInt(path.str());
 }
 
+IConstWUFileUsageIterator * CLocalWorkUnit::getFieldUsage() const
+{
+    CriticalBlock block(crit);
+    IPropertyTree* fieldUsageTree = p->queryPropTree("usedsources");
+
+    if (!fieldUsageTree)
+        return NULL;
+
+    IPropertyTreeIterator* iter = fieldUsageTree->getElements("*");
+    return new CConstWUFileUsageIterator(iter);
+}
+
+bool CLocalWorkUnit::getFieldUsageArray(StringArray & filenames, StringArray & columnnames, const char * clusterName) const
+{
+    bool scopeLoaded = false;
+    SCMStringBuffer defaultScope;
+
+    Owned<IConstWUFileUsageIterator> files = getFieldUsage();
+
+    if (!files)
+        return false; // this query was not compiled with recordFieldUsage option.
+
+    ForEach(*files)
+    {    
+        Owned<IConstWUFileUsage> file = files->get();
+
+        SCMStringBuffer filename;
+        file->getName(filename);
+
+        // filename cannot be empty
+        if (filename.length() < 3)
+            throw MakeStringException(WUERR_InvalidFieldUsage, "Invalid FieldUsage found in WU. Cannot enforce view security.");
+
+        // Filenames in field usage are surrounded with single quotes ('filename').
```

Comment:
Actually I understand why - because it is the ecl expression.

However the filenames may also be variable - in which case they are not known at compile time.  What should the code do in that situation?  It should verify that it is surrounded by quotes.  If not, should this fail?